src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadSubmissionResource.java

```diff
@@ -33,81 +45,314 @@
+        }
+
+        // Check the pattern
+        final var splittedFileName = file.getOriginalFilename().split("\\.");
+        final var fileSuffix = splittedFileName[splittedFileName.length - 1].toLowerCase();
+        final var filePattern = String.join("|", exercise.getFilePattern().toLowerCase().replace(" ", "").split(","));
+        if (!fileSuffix.matches(filePattern)) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .headers(HeaderUtil.createAlert(applicationName, "The uploaded file has the wrong type!", "fileUploadSubmissionIllegalFileType")).build();
+        }
+
+        final FileUploadSubmission submission;
+        try {
+            submission = fileUploadSubmissionService.handleFileUploadSubmission(fileUploadSubmission, file, exercise, principal);
+        }
+        catch (IOException e) {
+            return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert(applicationName, true, "fileUploadSubmission", "fileUploadSubmissionCantStore",
+                    "The uploaded file could not be saved on the server")).build();
+        }
+
+        hideDetails(submission);
+        return ResponseEntity.ok(submission);
     }
 
     /**
-     * PUT /file-upload-submissions : Updates an existing fileUploadSubmission.
+     * GET /file-upload-submissions/:id : get the "id" fileUploadSubmission.
      *
-     * @param fileUploadSubmission the fileUploadSubmission to update
-     * @return the ResponseEntity with status 200 (OK) and with body the updated fileUploadSubmission, or with status 400 (Bad Request) if the fileUploadSubmission is not valid, or
-     *         with status 500 (Internal Server Error) if the fileUploadSubmission couldn't be updated
-     * @throws URISyntaxException if the Location URI syntax is incorrect
+     * @param submissionId the id of the fileUploadSubmission to retrieve
+     * @return the ResponseEntity with status 200 (OK) and with body the fileUploadSubmission, or with status 404 (Not Found)
      */
-    @PutMapping("/file-upload-submissions")
-    public ResponseEntity<FileUploadSubmission> updateFileUploadSubmission(@RequestBody FileUploadSubmission fileUploadSubmission) throws URISyntaxException {
-        log.debug("REST request to update FileUploadSubmission : {}", fileUploadSubmission);
-        if (fileUploadSubmission.getId() == null) {
-            return createFileUploadSubmission(fileUploadSubmission);
-        }
-        FileUploadSubmission result = fileUploadSubmissionRepository.save(fileUploadSubmission);
-        return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, fileUploadSubmission.getId().toString())).body(result);
+    @GetMapping("/file-upload-submissions/{submissionId}")
+    public ResponseEntity<FileUploadSubmission> getFileUploadSubmission(@PathVariable Long submissionId) {
```

Comment:
`@PreAuthorize("hasAnyRole('USER','TA','INSTRUCTOR','ADMIN')")` is missing

Reply:
or `@PreAuthorize("hasAnyRole('TA','INSTRUCTOR','ADMIN')")` in case this should only be used by tutors and above

Reply:
Add `@PreAuthorize("hasAnyRole('TA','INSTRUCTOR','ADMIN')")`. User should not be able to acess it